public/app/plugins/datasource/elasticsearch/elastic_response.ts

```diff
@@ -424,40 +424,26 @@ export class ElasticResponse {
         throw this.getErrorFromElasticResponse(this.response, response.error);
       }
 
-      const hits = response.hits;
+      // We keep a list of all props so that we can create all the fields in the dataFrame, this can lead
+      // to wide sparse dataframes in case the scheme is different per document.
       let propNames: string[] = [];
-      let propName, hit, doc: any, i;
 
-      for (i = 0; i < hits.hits.length; i++) {
-        hit = hits.hits[i];
+      for (const hit of response.hits.hits) {
         const flattened = hit._source ? flatten(hit._source, null) : {};
-        doc = {};
-        doc[this.targets[0].timeField] = null;
-        doc = {
-          ...doc,
+        const doc = {
           _id: hit._id,
           _type: hit._type,
           _index: hit._index,
+          _source: { ...flattened },
```

Comment:
`flattened` is now both in doc root and in `_source` prop?

Reply:
it is the same as before, the code is just cleaned up a little

Reply:
It is the same as before just cleaned up a little. Before it was assigned to `_source` here https://github.com/grafana/grafana/pull/20290/files#diff-b2fa433b72195826bde52895d369871fL459

Reply:
Sorry, my bad didn't notice that.